### DISCUSSION TITLE
[Files storage] Custom local file system location

### DISCUSSION THREAD

**Discussion by cddqssc** [pocketbase/pocketbase#4631] [Q&A] [answered]
![image](https://github.com/pocketbase/pocketbase/assets/31430047/1a1897c9-51f1-447d-8132-5100d9af856a)
Can I customize the local path because I have mounted other disk devices?

Thanks!

**Comment by ganigeorgiev** [comment, marked as answer]
No, it is not possible at the moment (_I'll consider it maybe as a config option when extending PocketBase with Go_).

Depending on your use case and env setup an alternative could be to change the mount point of your disk to be `/path/to/pb_data/storage`.

**Comment by cddqssc** [comment]
Thank you for your quick reply. I am using network-mounted devices, and I am not sure if running PocketBase on these devices will cause performance issues (I suspect it might, due to the involvement of SQLite read and write operations). 
I would still prefer to add a config option as you mentioned!

**Reply by ganigeorgiev** [reply]
The internal SQLite db files (`pb_data/data.db` and `pb_data/logs.db`) are stored separately from the uploaded files (`pb_data/storage/*`). 

If all you want is to change the default location of the `pb_data` directory you can do that already with the `--dir` flag (you can explore all available global flags with `--help`).

If you change the mount point of your external drive to `pb_data/storage` then the database operations will be still performed on the original/local disk, aka. only the uploaded files will be stored on the externally mounted drive.

Note that the majority of the VPS providers that offers "Block storage", "Volumes", etc. operates with some kind of network mounted disk(s) and the performance in most cases based on my experience is acceptable. Here is an example benchmark comparing an external mounted Volume to the local disk for a Hetzner CX11 instance - https://pcr.cloud-mercato.com/providers/hetzner/flavors/cx11/performance/iops.

**Reply by cddqssc** [reply]
Thank you for your patience in answering!

I didn't make myself clear before.

I am using Hetner cloud servers, which have local disks that are NVMe, offering very fast access. I want PocketBase and the internal SQLite db files to run on the local NVMe(The default is already set up like this).

At the same time, my application has a very large demand for file storage (often several TBs).

So I cannot store these files on the local NVMe. I have purchased a 10TB Hetnzer storage box, and I would like to store files on it. 
However, the storage box currently does not compatible with S3, so I have to mount it as a network device to my cloud server, and I want to store files on it.

**Reply by ganigeorgiev** [reply]
I'm not familiar with the Hetzner storage box offering but if it can be mounted the same way as a regular external drive, then changing its mount point to `/path/to/pb_data/storage` should work, meaning that only the uploaded files will be stored on your external disk and the db file operations will remain on the local NVMe disk because they are stored outside of your external disk mount point (_I'm using a similar setup with Volumes for my other open source project as described in https://github.com/pocketbase/pocketbase/discussions/4254_).

**Reply by cddqssc** [reply]
This completely solved my problem, thank you very much. Additionally, if your read and write requirements for File storage are not high, you can try using Hetzner Storage Box, which is really cheap (€3.20/mo./1TB).
